### ISSUE TITLE
App mode Light/Dark/Windows default

### ISSUE THREAD

**Issue by myokeeh** [Microsoft/WindowsTemplateStudio#350]
It would be nice to have best practice Light/Dark/Windows default app mode settings baked into WTS to promote consistency.

This one's from the Maps app
![capture](https://cloud.githubusercontent.com/assets/3663297/26132870/032d9b40-3a58-11e7-9f0b-a7c2051652dc.PNG)

...among many!

**Comment by crutkas** [comment]
Fair, settings page will give you theme support but i do see that we don't respect the default.

would this be something you'd be interested in doing a PR for?  :)

**Comment by crutkas** [comment]
For the notes, i think this should be done by default for all apps

**Comment by degant** [comment]
Can I give this a shot? :)

**Comment by crutkas** [comment]
@degant 100%.  Remember to branch of dev.  I think this should be a pretty straight forward feature.  The Theme setting on the setting page may want to default to the OS theme then as well.

**Comment by Studio384** [comment]
I would say that there should also be taken a look at making this instant and not require users to restart the app. Skype does this, and if the theme is set to use the system color and it changes it works just fine, so why not allow users to switch the theme without restarting?

**Comment by mrlacey** [comment]
@Studio384 we will not be requiring a restart to change theme. (we don't currently either.)
Making it work without a restart is part of the challenge that's made this slow to implement.

**Comment by crutkas** [comment]
@studio384 I shared with Matt a POC. I think I have a working version.

**Comment by Studio384** [comment]
@mrlacey Had forgotten that WTS already does this without restart. The screenshot confused me. :)

**Comment by crutkas** [comment]
ok, think i got a near final POC.  Code behind but does everything, needs maybe a bit more TLC on setting page look feel but is functional now.

https://github.com/crutkas/Playground/tree/master/Issue350

**Comment by qbss** [comment]
I'm looking forward to this becoming active and maybe someone could correct this related minor issue (# 903) during the process.

**Comment by mrlacey** [comment]
The old XAML has been left commented out in the SettingsPage.xaml file and needs to be removed.

**Comment by crutkas** [comment]
Crap.  Will do today.  Good catch.

**Comment by mrlacey** [comment]
Thre's also a bug for Pivot&Tabs with CodeBehind - going to the settings tab resets to default

Repro:

- With Windows default theme set to Light on the PC/device
- New project with Pivot&Tabs and CodeBehind
- Add a settings page
- Run the generated app
- Go to the settings page and switch to dark theme
- Close app
- Reopen app (it should be in dark theme)
- Switch to settings tab

Expected:

- Selected item should display as dark

Actual:

- Selected item is shown as 'Windows Default'
- and theme switches back to the default (light theme)

**Comment by crutkas** [comment]
i can repro, why the code works on NavView but not Pivot is odd

**Comment by mrlacey** [comment]
The page lifecycle is different with the Pivot as the settings "page" is nested in a tab, rather than getting the normal page lifecycle events. That's where I'd look first. Having been bitten by similar issues in the past.

**Comment by crutkas** [comment]
think i have a fix, not to test

**Comment by crutkas** [comment]
Running unit tests and branch pushed. Will do PR once it is done.
